### PR TITLE
Sync in-memory store w/ Consul at constant interval

### PR DESCRIPTION
**Pull request by nathancoleman**
This is longer-term followup based on https://github.com/hashicorp/consul-api-gateway/issues/230#issuecomment-1157976119

### Changes proposed in this PR:
Trigger a sync for our in-memory store at a constant interval in the background

### How I've tested this PR:
_Still working on how to test this in a programmatic fasion_

### Checklist:
- [ ] Tests added
- [x] CHANGELOG entry added 
  > Run `make changelog-entry` for guidance in authoring a changelog entry, and
  > commit the resulting file, which should have a name matching your PR number.
  > Entries should use imperative present tense (e.g. Add support for...)

### CODE REVIEW

File: internal/commands/server/k8s_e2e_test.go [hunk 1]
```diff
@@ -299,7 +299,23 @@ func TestGatewayBasic(t *testing.T) {
...
+			// check for ingress gateway config entry
+			var entries []api.ConfigEntry
+			var err error
+			require.Eventually(t, func() bool {
+				entries, _, err = client.ConfigEntries().List(api.IngressGateway, &api.QueryOptions{Namespace: e2e.ConsulNamespace(ctx)})
+				return err == nil && len(entries) == 1
+			}, 5*time.Minute, checkInterval, "config entry not created in allotted time")
+
+			// verify background re-sync of config entries after delete
+			_, err = client.ConfigEntries().Delete(api.IngressGateway, entries[0].GetName(), &api.WriteOptions{Namespace: e2e.ConsulNamespace(ctx)})
```
Comment (andrewstucki) [inline internal/commands/server/k8s_e2e_test.go:311]: this is probably good enough, but if you want to exercise a bit more of the re-sync you could try and delete the gateway service from Consul directly. Deleting the service should remove all of its config entries as well, and we should have the service get re-registered via the registration goroutine in the deployment and the config entries subsequently re-created when the service appears again in the catalog.

Comment (nathancoleman) [reply internal/commands/server/k8s_e2e_test.go:311]: This sounds intriguing! Will dig into this today

File: internal/store/memory/store.go [hunk 2]
```diff
@@ -2,20 +2,22 @@ package memory
...
 	"github.com/hashicorp/consul-api-gateway/internal/core"
 	"github.com/hashicorp/consul-api-gateway/internal/metrics"
 	"github.com/hashicorp/consul-api-gateway/internal/store"
-	"github.com/hashicorp/go-hclog"
-	"github.com/hashicorp/go-multierror"
 )
 
 var (
-	ErrCannotBindListener = errors.New("cannot bind listener")
+	consulSyncInterval = 60 * time.Second
```
Comment (andrewstucki) [inline internal/store/memory/store.go:19]: we may want to consider moving this into a variable for command-line override eventually -- there should only be a few API calls per-gateway, but I would imagine that for some with large numbers of gateways deployed (not the normal case), this _could_ cause problems down the line if it's too aggressive. That said, 60s seems like it's fairly safe in like 99.9% of use-cases (i.e. non-overloaded Consul server or low number of gateways).

**Review by andrewstucki — APPROVED** [review]

File: internal/commands/server/k8s_e2e_test.go (hunk 1 above)
Comment (nathancoleman) [reply internal/commands/server/k8s_e2e_test.go]: After some digging, I discovered that deregistering the service from Consul does not remove config entries and intentions like we thought. Given that, I added additional assertions for both services and config entries so the tests are now more robust.

**Pull request merged**